pkg/controllers/deploy_mode_changer.go

```diff
@@ -206,3 +208,70 @@ func formatSaveOldDeployName(mc v1beta1.Milvus, component MilvusComponent) strin
 func formatSaveOldReplicaSetListName(mc v1beta1.Milvus, component MilvusComponent) string {
 	return fmt.Sprintf("%s-%s-old-replicas", component.Name, mc.Name)
 }
+
+// ChangeToOneDeployMode switches QueryNode to single deployment mode
+func (c *DeployModeChangerImpl) ChangeToOneDeployMode(ctx context.Context, mc v1beta1.Milvus) error {
+	if c.component != QueryNode {
+		return nil
+	}
+	oldDeploy, err := c.util.GetOldDeploy(ctx, mc, c.component)
+	if err != nil && !kerrors.IsNotFound(err) {
+		return errors.Wrap(err, "get old deployment for ChangeToOneDeployMode")
+	}
+	if oldDeploy != nil {
+		oldDeploy.Spec.Replicas = int32Ptr(0)
+		err = c.cli.Update(ctx, oldDeploy)
+		if err != nil {
+			return errors.Wrap(err, "scale down old deployment")
+		}
+		err = c.cli.Delete(ctx, oldDeploy)
```

Comment:
I think the Above `Update()` is redundant.  You can delete it directly, the pods will also be deleted.